src/together/utils/files.py

```diff
@@ -96,6 +96,143 @@ def check_file(
+            raise InvalidFileFormatError(
+                message=f"Found invalid role `{message['role']}` in the messages on the line {idx + 1}. "
+                f"Possible roles in the conversation are: {POSSIBLE_ROLES_CONVERSATION}",
+                line_number=idx + 1,
+                error_source="key_value",
+            )
+
+        if previous_role == message["role"]:
+            raise InvalidFileFormatError(
+                message=f"Invalid role turns on line {idx + 1} of the input file. "
+                "`user` and `assistant` roles must alternate user/assistant/user/assistant/...",
+                line_number=idx + 1,
+                error_source="key_value",
+            )
+        previous_role = message["role"]
+
+
+def validate_preference_openai(example: Dict[str, Any], idx: int = 0) -> None:
+    """Validate the OpenAI preference dataset format.
+
+    Args:
+        example (dict): Input entry to be checked.
+        idx (int): Line number in the file.
+
+    Raises:
+        InvalidFileFormatError: If the dataset format is invalid.
+    """
+    if not isinstance(example["input"], dict):
+        raise InvalidFileFormatError(
+            message="The dataset is malformed, the `input` field must be a dictionary.",
+            line_number=idx + 1,
+            error_source="key_value",
+        )
+
+    if "messages" not in example["input"]:
+        raise InvalidFileFormatError(
+            message="The dataset is malformed, the `input` dictionary must contain a `messages` field.",
+            line_number=idx + 1,
+            error_source="key_value",
+        )
+
+    validate_messages(example["input"]["messages"], idx)
+
+    for output_field in ["preferred_output", "non_preferred_output"]:
+        if not isinstance(example[output_field], list):
+            raise InvalidFileFormatError(
+                message=f"The dataset is malformed, the `{output_field}` field must be a list.",
+                line_number=idx + 1,
+                error_source="key_value",
+            )
+
+        if len(example[output_field]) != 1:
+            raise InvalidFileFormatError(
+                message=f"The dataset is malformed, the `{output_field}` list must contain exactly one message.",
+                line_number=idx + 1,
+                error_source="key_value",
+            )
+        if "role" not in example[output_field][0]:
+            raise InvalidFileFormatError(
+                message=f"The dataset is malformed, the `{output_field}` message is missing the `role` field.",
+                line_number=idx + 1,
+                error_source="key_value",
+            )
+        elif example[output_field][0]["role"] != "assistant":
+            raise InvalidFileFormatError(
+                message=f"The dataset is malformed, the `{output_field}` must contain an assistant message.",
+                line_number=idx + 1,
+                error_source="key_value",
+            )
+        
+
```

Comment:
Ultra-nit: two lines seems excessive; as per [PEP8](https://peps.python.org/pep-0008/#blank-lines), this is only needed for top-level definitions